src/pt/huntersscans/src/eu/kanade/tachiyomi/extension/pt/huntersscans/HuntersScans.kt

```diff
@@ -1,23 +1,247 @@
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
 import java.util.Locale
 import java.util.concurrent.TimeUnit
 
-class HuntersScans : Madara(
-    "Hunters Scan",
-    "https://huntersscan.xyz/",
-    "pt-BR",
-    SimpleDateFormat("dd 'de' MMMMM 'de' yyyy", Locale("pt", "BR")),
-) {
+class HuntersScans : ParsedHttpSource(), ConfigurableSource {
+    override val name = "Hunters Scans"
 
-    override val client: OkHttpClient = super.client.newBuilder()
-        .rateLimit(1, 2, TimeUnit.SECONDS)
-        .build()
+    override val lang = "pt-BR"
 
-    override val useNewChapterEndpoint = true
-    override val useLoadMoreRequest = LoadMoreStrategy.Always
+    override val supportsLatest = true
+
+    override val baseUrl = "https://huntersscan.xyz"
+
+    override val id = 7310576253330700902
+
+    private val mangaSubString = "manga"
+
+    private val preferences: SharedPreferences =
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+
+    override val client: OkHttpClient =
+        network.cloudflareClient.newBuilder()
+            .setRandomUserAgent(
+                preferences.getPrefUAType(),
+                preferences.getPrefCustomUA(),
+            )
+            .rateLimitHost(baseUrl.toHttpUrl(), 1, 2, TimeUnit.SECONDS)
+            .build()
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        addRandomUAPreferenceToScreen(screen)
+    }
+
+    override fun chapterFromElement(element: Element) = throw UnsupportedOperationException()
+
+    private fun fetchChapterRequest(url: HttpUrl, page: Int): List<SChapter> {
+        return try {
+            val mangaPaged = url.newBuilder()
+                .addQueryParameter("page", "$page")
+                .build()
+            chapterFromJScript(client.newCall(GET(mangaPaged, headers)).execute())
+        } catch (e: Exception) {
+            Log.e("HuntersScans", e.toString())
+            emptyList()
+        }
+    }
+
+    private fun chapterFromJScript(response: Response): MutableList<SChapter> {
+        return try {
+            val jScript = response.asJsoup().select(chapterListSelector())
+                .map { element -> element.html() }
+                .filter { element -> element.isNotEmpty() }
+                .first { chapterRegex.find(it) != null }
+
+            val chaptersLinks = chapterRegex.findAll(jScript)
+                .map { result -> result.groups.map { it?.value } }
+                .flatMap { it }
+                .toSet()
+
+            chaptersLinks.map { chapterLink ->
+                SChapter.create().apply {
+                    name = chapterLink!!.toChapterName()
+                    setUrlWithoutDomain(chapterLink.toChapterAbsUrl())
+                }
+            }.toMutableList()
+        } catch (e: Exception) {
```

Comment:
since you are catching any exception in `fetchChapterRequest` consider removing this here

Reply:
Resolved: https://github.com/keiyoushi/extensions-source/pull/1650/commits/d074c6330e3bf59ae82d36dbe0bfaa0d42f3392b